docs/specs/openapi.yaml

```diff
@@ -0,0 +1,335 @@
+        - services
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                action:
+                  type: string
+                  description: The action to perform. One of "start", "stop", "restart", "replan", "autostart".
+                  enum: ["start", "stop", "restart", "replan", "autostart"]
+                services:
+                  type: array
+                  description: |
+                    A list of service names.  Required for "start", "stop", and "restart".
+                    Ignored for "replan" and "autostart" (resolved automatically for "autostart" to default services).
+                  items:
+                    type: string
+            examples:
+              start one:
+                value:
+                  action: start
+                  services: ["svc1"]
+              stop multiple:
+                value:
+                  action: stop
+                  services: ["svc1", "svc2"]
+              restart multiple:
+                value:
+                  action: restart
+                  services: ["svc1", "svc2"]
+              replan:
+                value:
+                  action: replan
+              autostart:
+                value:
+                  action: autostart
+      responses:
+        '202':
+          description: Accepted - asynchronous operation started. Returns a change ID.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/PostServices"
+              examples:
+                accepted:
+                  value:
+                    type: async
+                    status-code: 202
+                    status: Accepted
+                    change: 1
+                    result: null
+        '400':
+          description: Bad Request - invalid request parameters.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorResponse"
+components:
+  schemas:
+    ErrorResponse:
+      type: object
+      properties:
+        type:
+          type: string
+          description: Response type, "error".
+        status-code:
+          type: integer
+          description: HTTP response status code.
+        status:
+          type: string
+          description: The text for the HTTP status code, see https://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml.
+        result:
+          $ref: "#/components/schemas/errorResult"
+    errorResult:
+      type: object
+      properties:
+        message:
+          type: string
+          description: Error message.
+        kind:
+          type: string
+          description: Error kind, possible values are "login-required", "no-default-services", "not-found", "permission-denied", "generic-file-error", "system-restart", and "daemon-restart".
```

Comment:
Ah yes, if we're moving the error description to their own section (not in the spec), let's list these possible kinds there.